ethcore/snapshot/src/service.rs

```diff
@@ -753,7 +751,8 @@ impl<C> Service<C> where C: SnapshotClient + ChainInfo {
 
 	/// Feed a chunk with the Restoration
 	fn feed_chunk_with_restoration(&self, restoration: &mut Option<Restoration>, hash: H256, chunk: &[u8], is_state: bool) -> Result<(), Error> {
-		let (result, db) = {
+		// todo[dvdplm] can re-work this to avoid the db clone (if it's indeed correct)
```

Comment:
is this about `rest.db.clone`? it is an `Arc`, so cloning it is cheap (one atomic increment).